worlds/noita/Items.py

```diff
@@ -66,9 +77,16 @@ def create_all_items(multiworld: MultiWorld, player: int) -> None:
         + create_kantele(multiworld.victory_condition[player])
     )
 
-    random_count = sum_locations - len(itempool)
-    itempool += create_random_items(multiworld, player, random_count)
+    # if there's not enough shop-allowed items in the pool, we can encounter gen issues
+    # 39 is the number of shop checks in the pool
+    shop_random_count = 0
+    if len(itempool) < 39:
+        shop_random_count = 39 - len(itempool)
+        random_count = locations_to_fill - 39
+    else:
+        random_count = locations_to_fill - len(itempool)
 
+    itempool += create_random_items(multiworld, player, shop_random_count, random_count)
```

Comment:
Maybe a little bit more convoluted than it needs to be. We can pass the dict to `create_random_items` instead of separate count values. Something like

```py
# 39 is the number of shop-valid items we need to guarantee
if len(itempool) < 39:
    itempool += create_random_items(multiworld, player, shop_only_filler_weights, 39 - len(itempool))
itempool += create_random_items(multiworld, player, item_filler_weights, locations_to_fill - len(itempool))
```

Reply:
`filler_weights` below can then be defined as I think something like

```py
filler_weights: Dict[str, int] = {
    **shop_only_filler_weights,
    "Gold (200)": 15,
    ....
```

Reply:
Doing it this way I'm getting constant failures with a minimal items solo yaml. The main issue: I don't think AP is placing the items correctly. I'm loading in exactly 7 perks, 32 hearts, and 7 Gold (200)s. The only way for it to work is if it places the perks + hearts in the 39 shop slots, and the 7 Gold (200)s in the 7 spell refresher spots. It is not doing this most of the time.

Reply:
Of note, the number of Golds it fails to place is not consistent. Most of the time, it fails to place 2 of them. Sometimes, 3. Sometimes, 1. Rarely, it succeeds.